runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPipelineRunner.java

```diff
@@ -67,14 +69,18 @@ public class FlinkPipelineRunner implements PortablePipelineRunner {
   public FlinkPipelineRunner(
       FlinkPipelineOptions pipelineOptions, @Nullable String confDir, List<String> filesToStage) {
     this.pipelineOptions = pipelineOptions;
-    this.confDir = confDir;
+    // confDir takes precedence than pipelineOptions.getFlinkConfDir
+    this.confDir = confDir != null ? confDir : pipelineOptions.getFlinkConfDir();
     this.filesToStage = filesToStage;
   }
 
   @Override
   public PortablePipelineResult run(final Pipeline pipeline, JobInfo jobInfo) throws Exception {
     MetricsEnvironment.setMetricsSupported(false);
 
+    // Apply log levels settings at the beginning of pipeline run
```

Comment:
do we need a similar change in https://github.com/apache/beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineRunner.java ? Can be  a separate PR.

Reply:
Yeah, not yet put it in because haven't tested it on spark runner. Entered #23713